### PR TITLE
feat(signature-collection): application api

### PR DESCRIPTION
**Pull request by alexdiljar**
# ...

Signature collection application apis and Initial query parameters for applications.

## What
- Adding signature collection client to the application api
- Documentation was missing instructions on how to use new client modules so added that
- Create signature collection list application template api
- Sign signature collection list template api.
- Functionality to pass initial parameters to applications through query params. This way templates can have initial parameters enabled and they will be added to answers when the application is created.

## Why

Initial query parameters are needed for signature collections. When a user has applied for signature collection lists for elections there will be a list created for each area, for the presidential elections there are 4 areas for sveitastjórnarkosningar there can be 50+. So that the electoral candidates dont have to have 4-50+ links to share with 
constituents we create one link with a query parameter with the electoral candidates id. When a user enters the to sign application we can use that id to figure out who's list they are trying to sign. 
Other applications will be able to pass a parameter to their application on creation with providing an  `initialQueryParameter` in the state config. 

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by peturgq — APPROVED** [review]
Infra looks good 👍

**Review by RunarVestmann — APPROVED** [review]
Codeowner files look good to me

File: libs/api/domains/signature-collection/src/lib/signatureCollection.service.ts [hunk 1]
```diff
@@ -81,14 +73,15 @@ export class SignatureCollectionService {
   }
 
   async signee(nationalId: string): Promise<SignatureCollectionSignee> {
-    return await this.signatureCollectionClientService.getSignee(nationalId)
+    const s = await this.signatureCollectionClientService.getSignee(nationalId)
```
[the remark this answers is not in the document]
Comment (alexdiljar) [reply libs/api/domains/signature-collection/src/lib/signatureCollection.service.ts:76]: I was using the const for a console log and forgot to replace, will fix!

**Review by rafnarnason — APPROVED** [review]
lgtm with a few minor comments

File: libs/application/template-api-modules/src/lib/modules/templates/signature-collection/signature-list-signing/signature-list-signing.service.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        default:
+          return errorMessages.deniedByService
+      }
+    })
+    throw new TemplateApiError(errors, 400)
+  }
+
+  async getList({ auth, application }: TemplateApiModuleActionProps) {
+    const areaId = (
+      (application.externalData.canSign.data as any)?.area as { id: string }
```
[the remark this answers is not in the document]
Comment (alexdiljar) [reply libs/application/template-api-modules/src/lib/modules/templates/signature-collection/signature-list-signing/signature-list-signing.service.ts]: The areaId is not needed here so I removed it, the service is checking for the users area.

**Review by obmagnusson — APPROVED** [review]
LGTM

**Review by saevarma — APPROVED** [review]
Core files LGTM

**Pull request merged**